sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -81,9 +81,35 @@ TEST(Context, BasicChar)
   EXPECT_TRUE(kind == ContextValue::ContextValueType::StdString);
 }
 
+TEST(Context, ApplicationContext)
+{
+  Context appContext = GetApplicationContext();
+
+  EXPECT_FALSE(appContext.HasKey("Key"));
+  EXPECT_FALSE(appContext.HasKey("key"));
+  EXPECT_FALSE(appContext.HasKey("Value"));
+  EXPECT_FALSE(appContext.HasKey("value"));
+  EXPECT_FALSE(appContext.HasKey("1"));
+  EXPECT_FALSE(appContext.HasKey(""));
+
+  auto duration = std::chrono::milliseconds(500);
+  EXPECT_FALSE(appContext.IsCancelled());
+  std::this_thread::sleep_for(duration);
```

Comment:
IMO, this test is kind of useless. 
Is it only checking that the context won't get cancelled after some time without doing anything to it? 

How is that different from setting some int to 1, then sleeping some time and checking that the int variable is still 1??

Maybe if you would add some threads to create other contexts and do some isCancell() checks during the sleep routine we would be adding some value to the test.

Reply:
I don't consider it useless.  
The scenario is different setting the value to 1 and then seeing if its still 1.
The scenario here checks that the context isn't cancelled.  It then waits for the duration and checks again to see if its cancelled.  If the Context internals worked correctly the context will now be in Cancelled state.

Calling IsCancelled() from other threads could be interesting as well.  Though it wouldn't remove the valid nature of this scenario

Reply:
>  It then waits for the duration and checks again to see if its cancelled.

But there is not any relation from `duration` to `appContext`, that's what I meant. `duration` is not added to context or something. That's why I consider this test similar to doing:
```cpp
int a = 1;
auto duration = std::chrono::milliseconds(500);
EXPECT_FALSE(a == 2);
std::this_thread::sleep_for(duration);
EXPECT_FALSE(a == 2);
```

If the test is for checking that the context won't auto-cancelled after 500 ms, then what if it auto-cancel at 501 ms?  or 1sec?  or 1 min?

Reply:
I agree, if we are talking about the whole unit test, this simple test as is, is useful. I wouldn't remove it. I also agree we should add the "cancel from another thread" test, in addition to this one.


> If the test is for checking that the context won't auto-cancelled after 500 ms, then what if it auto-cancel at 501 ms? or 1sec? or 1 min?

You bring up a good point @vhvb1989 , the `EXPECT_FALSE(appContext.IsCancelled());` at line 98, along with the sleep is meaningless here because that context wasn't created with a deadline.